test/Constraints/rdar62842651.swift

```diff
@@ -0,0 +1,12 @@
+// RUN: %target-swift-frontend -emit-sil -verify %s | %FileCheck %s
```

Comment:
Shouldn't this test be in test/SILGen then? (And it can use -emit-silgen instead of -emit-sil, which also runs mandatory passes)

Reply:
This is a problem where type-checker infers types _incorrectly_ I just could think of an easier and more robust way to verify it in test-case then check emitted SIL.

Reply:
I didn't want to match against constraint system output because that would just become a constant pain to maintain if anything changes with locators and or type variables.